public_html/app/controllers/Posts.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+    class Posts extends Controller {
+
+        public function __construct() {
+            if (!isLoggedIn()) {
+                redirect('users/login');
+            }
+
+            $this->postModel = $this->model('Post');
+        }
+
+
+        // INDEX
+        public function index() {
+            // Get posts
+            $posts = $this->postModel->getPosts();
+
+            $data = [
+                'posts' => $posts
+            ];
+
+            $this->view('posts/index', $data);
+        }
+
+
+        // ADD
+        public function add() {
+
+            if($_SERVER['REQUEST_METHOD'] == 'POST') {
+                // Sanitize POST array
+                $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+
+                $data = [
+                    'title' => trim($_POST['title']),
+                    'body' => trim($_POST['body']),
+                    'user_id' => $_SESSION['user_id'],
+                    'title_error' => '',
+                    'body_error' => ''
+                ];
+
+                // Validate entries
+                if(empty($data['title'])) {
+                    $data['title_error'] = 'Please enter title';
+                }
+                if(empty($data['body'])) {
+                    $data['body_error'] = 'Please enter body text';
+                }
+
+                // Make sure no errors
+                if(empty($data['title_error']) && empty($data['title_error'])) {
+                    // Validated
+                    if($this->postModel->addPost($data)) {
+                        flash('post_message', 'Post Added');
+                        redirect('posts');
+                    }
+                    else {
```

Comment:
Do not use nesting ifs. https://github.com/jupeter/clean-code-php#avoid-nesting-too-deeply-and-return-early-part-1